engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
 SECTION "Title screen", ROMX[$5D8C], BANK[$01]
 
 IntroSequence:: ; 5d8c
-    ; TODO
+	ld hl, GameFreakLogo ;E4000
+	ld a, $39
+	call FarCall_hl
+	jr c, TitleSequenceStart ;5dae
+	ld a, [wOpeningType] ;cc38
+	and a
+	jr z, .opening_sequence_start
+	
+.pikachu_minigame_start
+	ld hl, PikachuMiniGame ;E0000
+	ld a, $38
+	call FarCall_hl
+	jr TitleSequenceStart ;5dae
 
+.opening_sequence_start
+	ld hl, OpeningSequence ;E432F
+	ld a, $39
+	call FarCall_hl
+
+TitleSequenceStart::
+	call TitleSequenceInit
+	ld hl, SetTitleBGDecorationBorder ;91D2
+	ld a, $02
+	call FarCall_hl
+	
+.loop
+	call TitleScreenMain ;5fb8
+	jr nc, .loop
+	
+	call ClearPalettesAndWait ;361e -clear palettes and wait
+	call ClearSprites
+	ld a, $01
+	ldh [$FFDE], a
+	call ClearTileMap
+	call ResetPalette ;33a
+	
+	ld a, [wSequence + 1]	; wSequence + 1 = $CB5F
+	ld e, a
+	ld d, $00
+	ld hl, TitleScreenJumpTable
+	add hl, de
+	add hl, de
+	ldi a, [hl]
+	ld h, [hl]
+	ld l, a
+	
+	jp hl
+
+TitleScreenJumpTable::
+	dw MainMenu
+	dw DebugMenu ;4031
+	dw SRAMClearMenu ;61c6
+	dw IntroSequence
+
+TitleSequenceInit::
+	call ClearPalettes
+	
+	xor a
+	ldh [$FFE8], a
+	ldh [$FFDA], a
+	ldh [$FFD9], a
+	
+	ld de, $0000 ;Stop the music.
+	call PlayMusic
+	
+	call ClearTileMap
+	call DisableLCD
+	call ClearSprites
+	
+	ld a, $23
+	ld hl, InitEffectObject	; 8CCFD - create another object?
+	call FarCall_hl
+	ld hl, $8000
+	ld bc, $1800
+
+.clear_loop
+	ld [hl], $00
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .clear_loop
+
+	ld hl, TitleScreenGFX	; TitleScreenGFX = $47CF
+	ld de, $9410
+	ld bc, $00D0
```

Comment:
`ld bc, 13 tiles`

Reply:
Done. But how do you calculate 13 tiles from $00D0? Just making sure.

Reply:
13 = 0x0D
1 tile is 0x10 bytes
13 * 0x10 = 0x0D * 0x10 = 0xD0
"tiles" is a constant string expression which basically adds " * 0x10" where it's placed
so `ld bc, 13 tiles` turns into `ld bc, 13 * 0x10`